File: ozomall_wepy/src/api/goods.js
import service from '../utils/service'

// 获取商品列表
export function goodsList(params) {
    return service({
        url: '/mall/goods/list',
        method: 'get',
        params
    })
}

// 搜索商品
export function searchGoods(params) {
    return service({
        url: '/mall/goods/search',
        method: 'get',
        params
    })
}

// 获取商品信息
export function getGoods(params) {
    return service({
        url: '/goods/get',
        method: 'get',
        params
    })
}

// 获取商品数量
export function getGoodsCount(params) {
    return service({
        url: '/mall/goods/getGoodsCount',
        method: 'get',
        params
    })
}

// 获取商品图片
export function getGoodsPic(params) {
    return service({
        url: '/goods/getGoodsPic',
        method: 'get',
        params
    })
}

// 获取商品规格
export function getGoodsAttr(params) {
    return service({
        url: '/goods/getGoodsAttr',
        method: 'get',
        params
    })
}

// 获取商品价格
export function getGoodsSkuList(params) {
    return service({
        url: '/goods/getGoodsSkuList',
        method: 'get',
        params
    })
}

// 获取商品参数
export function getGoodsParams(params) {
    return service({
        url: '/goods/getGoodsParams',
        method: 'get',
        params
    })
}

File: ozomall_flutter/lib/api/goods.dart
import 'package:ozomall_flutter/utils/server.dart';

class GoodsApi {
  // 获取商品列表
  static Future getGoodsList(params) {
    return Server.get("/mall/goods/list", params);
  }

  // 获取商品详情
  static Future getGoodsDetail(params) {
    return Server.get("/goods/get", params);
  }

  // 获取商品图片
  static Future getGoodsPic(params) {
    return Server.get("/goods/getGoodsPic", params);
  }

  // 获取商品参数
  static Future getGoodsParams(params) {
    return Server.get("/goods/getGoodsParams", params);
  }

  // 获取商品品牌信息
  static Future getGoodsBrandInfo(params) {
    return Server.get("/goods/getGoodsBrandInfo", params);
  }

  // 根据品牌获取商品数量
  static Future getGoodsCount(params) {
    return Server.get("/mall/goods/getGoodsCount", params);
  }

  // 获取商品属性
  static Future getGoodsAttr(params) {
    return Server.get("/goods/getGoodsAttr", params);
  }

   // 获取商品价格
  static Future getGoodsSkuPrice(params) {
    return Server.get("/goods/getGoodsSkuList", params);
  }
}
